src/core/user_operation.ml

```diff
@@ -1,12 +1,16 @@
 open Helpers
 open Crypto
 type initial_operation =
-  | Transaction    of {
+  | Transaction          of {
       destination : Key_hash.t;
       amount : Amount.t;
       ticket : Ticket_id.t;
     }
-  | Tezos_withdraw of {
+  | Contract_origination of {
+      to_originate : Smart_contracts.Origination_payload.t;
```

Comment:
Origination_payload should be defined here. Also call it just `Contract`

Reply:
I'd keep it in smart_contracts.ml for simplicity now